### PR TITLE
Added a how-to guide for using conda for development dependencies

### LINKED ISSUE #6445: Provide guidance on setting up a development environment with all dependencies installed through `conda`?
Currently, our developer's guide tells users to use `conda` to set up an environment and install `pip`, and then use `pip` rather than `conda` to install all of `sunpy`'s dependencies.  If a user subsequently uses `conda` to update any dependencies in that environment, that environment may become broken for further `conda` use.

We could consider providing guidance on how to best use a local repo for `sunpy` but still use  `conda` for all of the dependencies, e.g.:
```
conda install sunpy
conda remove --force sunpy
pip install --no-build-isolation --no-deps -e .
```

### PR DESCRIPTION
**Pull request by ayshih**
Closes #6445

### CODE REVIEW

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+****************************************
+Using conda for Development Dependencies
+****************************************
+
+The ``conda`` package for ``sunpy`` specifies only those dependencies that are relevant for the typical user.
+A power user or a developer will instead need the complete set of dependencies to use all optional features, build the documentation, and/or run the full test suite.
+We provide a ``conda`` environment file (``sunpy-dev-env.yml``) in the base directory of the GitHub repository.
+You can create a ``conda`` environment (here named ``sunpy-dev``) from this file:
+
+.. code:: bash
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:12]:
```suggestion
.. code-block:: bash
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    $ conda env create -n sunpy-dev -f sunpy-dev-env.yml
+    $ conda activate sunpy-dev
+
+Alternatively, if you want to add the complete set of dependencies to an existing ``conda`` environment, you can use this file to update that environment (here named ``existing-env``):
+
+.. code:: bash
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:19]:
```suggestion
.. code-block:: bash
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    $ conda env update -n existing-env -f sunpy-dev-env.yml
+
+.. note::
+
+    This conda environment file intentionally does not specify restrictions on release versions for any dependency.
+
+This ``conda`` environment file specifies only the *dependencies* for ``sunpy``, and not ``sunpy`` itself.
+Depending on your needs, continue with one of the following two ways to install ``sunpy`` in this ``conda`` environment.
+
+
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:30]:
```suggestion
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Normal Installation of sunpy
+============================
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:32]:
```suggestion
Normal Installation of ``sunpy``
================================
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+If you do not plan to modify the code of ``sunpy`` itself, you can simply install ``sunpy`` via ``conda``:
+
+.. code:: bash
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:36]:
```suggestion
.. code-block:: bash
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    $ conda install sunpy
+
+Since the ``conda`` environment already has the complete set of dependencies for ``sunpy``, this call should install only ``sunpy`` and no additional packages.
+
+
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:42]:
```suggestion
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+Editable Installation of sunpy
+==============================
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:44]:
```suggestion
Editable Installation of ``sunpy``
==================================
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+.. code:: bash
+
+    $ pip install --no-deps --no-build-isolation -e .
+
+The ``--no-deps`` and ``--no-build-isolation`` options ensure that ``pip`` does not itself install any dependencies.
+Since the ``conda`` environment is designed to already have the complete set of dependencies, the ``pip`` installation should succeed.
+
+You now have a ``conda`` environment with an editable installation of ``sunpy`` and with (nearly) all dependencies managed by ``conda``.
+As you install other packages in this environment, a package that depends on ``sunpy`` will trigger ``conda`` to install ``sunpy``.
+*That is fine!*
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:63]:
```suggestion
**That is fine!**
```
I think bold has more emphasis

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+This ``conda`` installation of ``sunpy`` will simply mask the ``pip`` installation of ``sunpy``.
+All you need to do is to remove the ``conda`` installation with the ``--force`` option so that dependencies are left undisturbed:
+
+.. code:: bash
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:67]:
```suggestion
.. code-block:: bash
```

File: docs/dev_guide/contents/conda_for_dependencies.rst [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+Once the ``conda`` installation of ``sunpy`` is removed, the ``pip`` installation of ``sunpy`` will automatically be accessible again.
+
+.. note::
+
+    For those who use ``mamba`` instead of ``conda``, most ``conda`` commands can be translated by simply substituting "mamba" for "conda".  However, ``mamba remove`` does not support the ``--force`` option, so you do in fact have to call ``conda remove``.
+
+As a tip, you can follow a similar procedure to incorporate editable installations of other packages (e.g., ``astropy``) in a ``conda`` environment.
+You first install the package via ``conda`` to ensure its dependencies are present, then you remove the package alone without disturbing the dependencies, and finally you perform the editable install of the package from the base directory of your local repository:
+
+.. code:: bash
```
Comment (nabobalis) [inline docs/dev_guide/contents/conda_for_dependencies.rst:80]:
```suggestion
.. code-block:: bash
```

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Probably the most useless "review" so I approved it regardless.

**Comment by nabobalis** [comment]
I leave it up to you if you want a changelog entry. If not I will merge it.

**Comment by ayshih** [comment]
I'll note that the Newcomer's Guide, under the Developer's Guide, uses `code` blocks instead of `code-block` blocks (at time of this writing).  I happened to pick that file to look at to see which one we used.

**Comment by nabobalis** [comment]
> I'll note that the Newcomer's Guide, under the Developer's Guide, uses `code` blocks instead of `code-block` blocks (at time of this writing). I happened to pick that file to look at to see which one we used.

Oh I plan to change that as part of the docs overhaul, I think I converted them in my draft docs PR.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**